### ISSUE TITLE
The Copy button of the PostURL shouldn't be placed in the slug input

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63700]
### Description

Noticed while working on https://github.com/WordPress/gutenberg/issues/61196 / https://github.com/WordPress/gutenberg/pull/63669

In the PostURL popover, there is a `Copy` button that is visually placed within the slug input field. As such, it is visually associated with the slug value. Screenshot:

![Screenshot 2024-07-18 at 13 07 48](https://github.com/user-attachments/assets/0e6dd3cf-9ac3-4160-92f1-bc43d0e860d0)

As a user, I would expect the Copy button to copy the value that is inside the input field, because that's what the UI is showing me.

Instead, the Copy button actually copies the full permalink that is shown _below_ the input field.

Following the example in the screenshot above:
- The input field shows `/my-post-title`.
- The copy button is labeled `Copy` and is visually associated with the input field.
- Clicking the button copies the full permalink `http://localhost:8889/my-post-title/` instead.

I'm not sure this is a good UI as it is potentially confusing for users. Also, it is inconsistent with other UIs where the Copy button actually copies the value shown in the input field.

### Step-by-step reproduction instructions

See above.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by afercia** [comment]
In https://github.com/WordPress/gutenberg/pull/57168#issuecomment-1938020796 @SaxonF pointed out a few interesting considerations including:

> A fairly standard pattern is more closely mapping copy icon to the actual value being copied, either within the input for example or just outside. 

In this cas, 'the actual value being copied' isn't the one of the input. It's the full permalink shown below the input.

I'd appreciate some design feedback @WordPress/gutenberg-design

**Comment by richtabor** [comment]
> In this cas, 'the actual value being copied' isn't the one of the input. It's the full permalink shown below the input.

Yes, it's not ideal having the copy button copy the whole link, but next to the slug. This should be cleaned up. Maybe the copy icon to the right of the full URL. And a snack bar notification confirming the copy.

**Comment by t-hamano** [comment]
There have been some suggestions for changes to the location of the copy button.

- https://github.com/WordPress/gutenberg/issues/61196#issuecomment-2241610484
- https://github.com/WordPress/gutenberg/pull/63706#issuecomment-2239417982
- https://github.com/WordPress/gutenberg/issues/56381#issuecomment-1843255480
- https://github.com/WordPress/gutenberg/issues/56381#issuecomment-1847685455
